src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -475,5 +476,54 @@ private static bool SupportsCSharpCodeActions(JToken token)
 
             return Task.FromResult(result.ToArray());
         }
+
+        public override async Task<VSInternalWrapWithTagResponse> RazorWrapWithTagAsync(VSInternalWrapWithTagParams wrapWithParams, CancellationToken cancellationToken)
+        {
+            // Same as in LanguageServerConstants, and in Web Tools
+            const string HtmlWrapWithTagEndpoint = "textDocument/_vsweb_wrapWithTag";
+
+            var response = new VSInternalWrapWithTagResponse(wrapWithParams.Range, Array.Empty<TextEdit>());
+
+            var hostDocumentUri = wrapWithParams.TextDocument.Uri;
+            if (!_documentManager.TryGetDocument(hostDocumentUri, out var documentSnapshot))
+            {
+                return response;
+            }
+
+            string languageServerName;
+            Uri projectedUri;
+            if (documentSnapshot.TryGetVirtualDocument<HtmlVirtualDocumentSnapshot>(out var htmlDocument))
+            {
+                languageServerName = RazorLSPConstants.HtmlLanguageServerName;
+                projectedUri = htmlDocument.Uri;
+            }
+            else
+            {
+                Debug.Fail("Unexpected RazorLanguageKind. This can't really happen in a real scenario.");
+                return response;
```

Comment:
Just curious why we landed on returning an empty array for these error cases?

Reply:
Just copying the semantics of the Web Tools implementation, since they're the ones that handle the response. Null should work too though, if you think it would be better?

Reply:
Nope, don't want to rock the boat (at least at this juncture), it just feels like our various implementations are inconsistent about when to use null vs empty-return vs error return status. Not something to solve in this PR though.